src/lib.rs

```diff
@@ -30,20 +30,29 @@
 //! ```toml
 //! ada-url = { version = "1", features = ["serde"] }
 //! ```
+//!
+//! ## no-std
+//!
+//! Whilst `ada-url` has `std` feature enabled by default, you can set `no-default-features`
+//! get a subset of features that work in no-std environment.
+//!
+//! ```toml
+//! ada-url = { version = "1", no-default-features = true }
+//! ```
+
+#![cfg_attr(not(feature = "std"), no_std)]
 
 pub mod ffi;
 mod idna;
 pub use idna::Idna;
 
 use core::{borrow, ffi::c_uint, fmt, hash, ops};
-use derive_more::{Display, Error};
-
-#[cfg(feature = "serde")]
```

Comment:
Why do we not need this anymore?

Reply:
You are referring to the following block?

```rust
#[cfg(feature = "serde")]
extern crate serde;
```

The answer is: It's not necessary to begin with.

Reply:
`extern crate` is only required in edition 2015, edition 2021 does not require them.